### PR TITLE
add support for dapp-runner's ingress command execution

### PR DESCRIPTION
**Pull request by shadeofblue**

### CODE REVIEW

File: dapp_manager/dapp_manager.py [hunk 1]
```diff
@@ -88,6 +92,40 @@ def read_file(self, file_type: RunnerFileType, ensure_alive: bool = True) -> str
...
+        m = re.match("^(?P<service_name>.*?)(\\[(?P<idx>\\d+)\\])?$", service)
+        if not m:
+            raise ValueError("`service` parameter format unknown, use ")
+        service_name = m.group("service_name")
+        service_idx = int(m.group("idx") or 0)
+        command_msg = json.dumps({service_name: {service_idx: command}})
+        start = datetime.now()
+        with self.storage.open("data", "r") as data:
+            data.seek(0, 2)
+            self.storage.write_file("commands", command_msg)
```
Comment (lucekdudek) [inline dapp_manager/dapp_manager.py:106]: I think this `self.storage.write_file("commands", command_msg)` can be moved to line 103
```python
102        command_msg = json.dumps({service_name: {service_idx: command}})
103        self.storage.write_file("commands", command_msg)
```

File: dapp_manager/dapp_manager.py [hunk 2]
```diff
@@ -88,6 +92,40 @@ def read_file(self, file_type: RunnerFileType, ensure_alive: bool = True) -> str
...
+                        print(cdict.get("command"))
+                        print("success: ", cdict.get("success"), "\n")
+                        print(cdict.get("stdout"))
+                        stderr = cdict.get("stderr")
+                        if stderr:
+                            print("\nstderr:")
+                            print(stderr)
+                    break
+
+                raise TimeoutError()
```
Comment (lucekdudek) [inline dapp_manager/dapp_manager.py:127]: I would split that into more methods for more readability
```python
def exec_command(self, service: str, command: List[str], timeout: int):
     self._ensure_alive()
     ... = eval_input()
     while ... :
         ... = read_msg()
         ... = foo() # last part which IDK if does anything more than `print`
 ```

**Review by lucekdudek — COMMENTED** [review]
Looks good. Added two suggestions to `def exec_command`,

File: dapp_manager/dapp_manager.py (hunk 2 above)
Comment (shadeofblue) [reply dapp_manager/dapp_manager.py]: @lucekdudek I updated the function by pulling out the input and output part -> I don't feel it's necessary to split-out the very essence of it...

File: dapp_manager/dapp_manager.py (hunk 1 above)
Comment (lucekdudek) [reply dapp_manager/dapp_manager.py:124]: How about 124
```python
123        command_msg = json.dumps({service_name: {service_idx: command}})
124        self.storage.write_file("commands", command_msg)
```

**Review by lucekdudek — APPROVED** [review]

**Pull request merged**